src/main/java/ru/mail/polis/homework/objects/MaxTask.java

```diff
@@ -13,43 +13,80 @@ public class MaxTask {
      * НЕЛЬЗЯ СОРТИРОВАТЬ массив array и его копии
      * 4 тугрика
      */
-    public static int[] getMaxArray(int[] array, int count) {
-        if (array == null || array.length < count) {
-            return null;
-        }
-        int copyLength = array.length;
-        int[] arrayCopy = Arrays.copyOf(array, copyLength);
-        int[] max = new int[count];
-        int indexMaxValue = -1;
-        int[] lastMaxValue = new int[count];
-        for (int i = 0; i < lastMaxValue.length; i++) {
-            lastMaxValue[i] = -1;
-        }
-        for (int j = 0; j < count; j++) {
-            int i = 0;
-            int tmpMax = Integer.MIN_VALUE;
-            while (i < arrayCopy.length) {
-                boolean flag = false;
-                for (int k = 0; k < lastMaxValue.length; k++) {
-                    if (i == lastMaxValue[k]) {
-                        flag = true;
+    public static void binarySearchIndex(int[] a, int item) {
+        int low = 0, high = a.length - 1;
+        if (item >= a[low]) {
+            offsetArray(a, low);
+            a[low] = item;
+        } else if (item > a[high]) {
+            while (low <= high) {
+                int mid = low + ((high - low) / 2);

```

Comment:
порядок действий же, скобки не нужны
```suggestion
                int mid = low + (high - low) / 2;

```